Dataset.py

```diff
@@ -291,11 +305,13 @@ def get_num_timesteps(self):
     """
     :rtype: int
     """
-    assert self._num_timesteps > 0
+    assert self._num_timesteps > 0 # TODO: this can only be valid for HDFDataset
     return self._num_timesteps
 
   def get_num_codesteps(self):
     """
+    TODO: this function has no usages
```

Comment:
I remember that I have seen usages in some user scripts (for dumping HDF or so - external, not part of RETURNN). But not sure. Maybe I remember wrongly. Maybe ask @doetsch or @pavelgolik . Or maybe remove and see if someone complains.